app/src/features/Sources/sourceSlice.ts

```diff
@@ -0,0 +1,62 @@
+import { createSlice, createAction } from "@reduxjs/toolkit";
+
+import type { RootState } from "app/store";
+import type { Source, Credential } from "services/api/generated/api.generated";
+
+export enum EditTypeEnum {
+  Source = "SOURCE",
+  Credential = "CREDENTIAL",
+  Owners = "OWNERS",
+}
+
+type SourceSliceState = {
+  current?: Source;
+  credential?: Credential;
+  editType?: EditTypeEnum;
+};
+
+const initialState: SourceSliceState = {};
+
+export const editSource = createAction<Source>("editSource");
+export const sourceEdited = createAction<Source>("sourceEdited");
+export const credentialEdited = createAction<Credential>("credentialEdited");
```

Comment:
Forgot to address this ! I will look into it!

Reply:
Actually I feel like it may be a nice design, as it lets you expose "main/more important" actions to other slices if needed

Reply:
I think for now on I'm just going to put actions with payload in `extraReducers` and the rest in `reducers`. The doc says it may be recommended to use the builder api with Typescript and I understand it's related to the typing of the payload..